### PR TITLE
Introduce a helper returning the system CPUs

### PR DESCRIPTION
**Pull request by xvallspl**
...refactoring the old way of obtaining it.

Just for convenience.

### CODE REVIEW

File: core/base/inc/TROOT.h [hunk 1]
```diff
@@ -95,6 +95,9 @@ namespace ROOT {
    void DisableImplicitMT();
    Bool_t IsImplicitMTEnabled();
    UInt_t GetImplicitMTPoolSize();
+
+   /// Get the number of system CPUs.
+   UInt_t GetNCPUs();
```
Comment (pcanal) [inline core/base/inc/TROOT.h]: (Especially since) this is not configurable, this routine belongs in TSystem.  Thanks.

Comment (xvallspl) [reply core/base/inc/TROOT.h]: Sure!

**Comment by dpiparo** [comment]
Hi @xvallspl , what problem is this change solving?

**Comment by xvallspl** [comment]
It's a proposal, totally open to discussion.

I found myself in need of this piece of code several times and I always thought it would be cool to have it as a single call instead of having to remember those three lines.

You can see it a nicer and useful way for writing nCPUs dependent, processor-agnostic code. I would say it's better than replicating the current process of getting the number of CPUs all over the place.

**Comment by xvallspl** [comment]
Moved it to TSystem.

**Comment by xvallspl** [comment]
Just remembered std::thread::hardware_concurrency. Closing the PR.

**Pull request closed without merging**